model/src/main/kotlin/PackageReference.kt

```diff
@@ -98,7 +98,12 @@ data class PackageReference(
      * A comparison function to sort package references by their identifier. This function ignores all other properties
      * except for [id].
      */
-    override fun compareTo(other: PackageReference) = id.compareTo(other.id)
+    override fun compareTo(other: PackageReference) =
+        when {
+            id < other.id -> -1
+            id > other.id -> 1
+            else -> if (this == other) 0 else 1 // Return arbitrary inequality.
```

Comment:
The changes in this commit do impact the performance negatively.

In this particular case it leads to recursion: the whole subtree rooted at `this` and `other` is compared, which I doubt is a good idea.

Reply:
> the whole subtree rooted at `this` and `other` is compared, which I doubt is a good idea.

Actually, that's exactly the point in order to be able to use Kotest's data class matchers e.g. on whole OrtResults. This would allow to get rid of the fragile text line comparison which is really hard to grasp on CI.

Reply:
But the text comparison also makes it very easy to see the effect of changes during development and to update the expected results files, if the plan is to change the matching for the functional analyzer tests we need to consider this.

Reply:
> But the text comparison also makes it very easy to see the effect of changes

True, and it makes it easy to take over new expected results from actual results. My main point is that I believe Kotest's data class matchers *should* work with out code base in case someone decides to use them in tests, but currently they don't as soon as a `SortedSet` is involved anywhere in the hierarchy.